### PR TITLE
Remove DLS module and move things into1 single flowredux module

### PR DESCRIPTION
**Pull request by sockeqwe**
Addresses #270 : Merges `dsl` module into `flowredux` module so that only 1 modules lives along with the `compose` module.

Open points:
- [x] Merge code (incl. tests) from dsl module into flowredux module
- [x] make internal API internal (modifiers)
- [x] Remove duplicated code such as `TestUtils` etc.
- [x] update README

Open question: Do we also want to rename packages? I think keeping things under `flowredux.dsl` is not too bad to have the possibility to add things somewhen in later versions directly under `flowredux` package. Also doesn't break backward compatibility.

### CODE REVIEW

**Comment by gabrielittner** [comment]
I think keeping the package is fine 👍

File: flowredux/src/commonMain/kotlin/com/freeletics/flowredux/SideEffect.kt [hunk 1]
```diff
@@ -11,10 +11,10 @@ import kotlinx.coroutines.flow.Flow
  * @param state [GetState] to get the latest state of the state machine
  */
 // TODO find better name?
```
Comment (gabrielittner) [inline flowredux/src/commonMain/kotlin/com/freeletics/flowredux/SideEffect.kt:13]: Should we also remove this todo?

**Comment by sockeqwe** [comment]
@gabrielittner  ready for review now. Could you please take a look? Maybe I have missed something

File: flowredux/src/commonMain/kotlin/com/freeletics/flowredux/SideEffect.kt (hunk 1 above)
Comment (sockeqwe) [reply flowredux/src/commonMain/kotlin/com/freeletics/flowredux/SideEffect.kt]: good catch 👍  

Removed

**Review by gabrielittner — APPROVED** [review]
I didn't find anything that's left 👍

**Comment by sockeqwe** [comment]
I will create a new release now with these changes

**Pull request merged**